sourceme.sh

```diff
@@ -1,10 +1,24 @@
 #!/bin/sh
 # Convenience script for setting variables for in-tree use of hammer.
-if [ -z "$HAMMER_HOME" ]; then
-    >&2 echo "Must set HAMMER_HOME to root. Try export HAMMER_HOME="\$PWD" for the current dir."
+
+if [ -z "${HAMMER_HOME}" ]; then
+  # Try to find the location of the script even if it's called through a symlink.
+  #  https://stackoverflow.com/questions/59895/how-to-get-the-source-directory-of-a-bash-script-from-within-the-script-itself
+  SOURCE="${BASH_SOURCE[0]}"
```

Comment:
This seems pretty complicated for what I would think would be a fairly simple thing. Does
```
dirname ${BASH_SOURCE[0]} | xargs readlink -f
```
 not work?

Reply:
some version of this does work for the platforms we care about and the invocation methods we are most likely to use (linux, bash, and one of sourcing or executing the script); much nicer, thanks